aten/src/ATen/native/SharedReduceOps.h

```diff
@@ -319,15 +319,29 @@ namespace detail {
 
 template <typename scalar_t>
 struct LessOrNan {
-  C10_DEVICE bool operator () (scalar_t a, scalar_t b) const {
-    return at::_isnan(a) || a < b;
+  C10_DEVICE bool operator () (scalar_t a, scalar_t b, int64_t idx_a, int64_t idx_b) const {
+    // If (a == b), then choose the one with lower idx, else min(a, b)
+    if (at::_isnan(a)) {
+      if (at::_isnan(b)) {
```

Comment:
Sorry, I know I suggested this previously but now that I have more time do we need to know which NaN to return since we're not support complex values? That is, can this check just be

```
if (at::_isnan(a)) {
  return true;
}
```

My mistake on not asking this earlier.

Reply:
Ah! It's because the indices are used later. My mistake x2.

Reply:
Yes, this is to match numpy's behaviour of returning the index of first ocurrence `nan` in case there are multiple nans present.

Reply:
> Yes, this is to match numpy's behaviour of returning the index of first ocurrence `nan` in case there are multiple nans present.

You're totally right, of course. I was just looking at too many issues ;)